scripts/copyFiles.mjs

```diff
@@ -89,17 +89,27 @@ async function createPackageFile() {
   const { nyc, scripts, devDependencies, workspaces, ...packageDataOther } =
     JSON.parse(packageData);
 
+  const topLevelPathImportsAreCommonJSModules = fse.existsSync(
+    path.resolve(buildPath, './esm/index.mjs'),
+  );
+
   const newPackageData = {
     ...packageDataOther,
     private: false,
     ...(packageDataOther.main
       ? {
-          main: fse.existsSync(path.resolve(buildPath, './node/index.js'))
-            ? './node/index.js'
-            : './index.js',
-          module: fse.existsSync(path.resolve(buildPath, './esm/index.js'))
-            ? './esm/index.js'
-            : './index.js',
+          main: topLevelPathImportsAreCommonJSModules ? './index.js' : './node/index.js',
```

Comment:
That's aside from the main purpose of this PR, but since we're touching buil outputs we may as well consider it:
I find it weird to have an esm build in a subdirectory of node build (or vice versa). Could we have top-level directories for all build outputs and have the exports field pointing to each one?

Reply:
Maybe it makes sense to start a shaping page to discuss package layout? I'd even consider one flat folder with all targets together, separately identifiable by extension.